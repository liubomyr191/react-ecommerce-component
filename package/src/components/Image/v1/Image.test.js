import React from "react";
import renderer from "react-test-renderer";
import Image from "./Image";

test("Basic Image snapshot", () => {
  const component = renderer.create((
    <Image src="/reaction-design-system-logo.svg" alt="Reaction Design System Logo" />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Image with progressive loading snapshot", () => {
  const component = renderer.create((
    <Image
      src="/images/responsive-sticker/medium.jpg"
      presrc="/images/responsive-sticker/small.png"
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Responsive Image with progressive loading snapshot", () => {
  const component = renderer.create((
    <Image
      presrc="/images/responsive-sticker/small.png"
      srcs={{
        large: "/images/responsive-sticker/large.jpg",
        medium: "/images/responsive-sticker/medium.jpg",
        small: "/images/responsive-sticker/small.png"
      }}
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
