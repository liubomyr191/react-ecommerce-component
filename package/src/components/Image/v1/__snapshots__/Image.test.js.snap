// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Image snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  display: block;
  height: 0;
  overflow: hidden;
  padding-top: 100%;
  position: relative;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`Image with progressive loading snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  display: block;
  height: 0;
  overflow: hidden;
  padding-top: 100%;
  position: relative;
  width: 100%;
}

.c1 {
  height: auto;
  left: 50%;
  opacity: 1;
  position: absolute;
  -webkit-transition: opacity 300ms cubic-bezier(0.4,0,0.2,1);
  transition: opacity 300ms cubic-bezier(0.4,0,0.2,1);
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  -webkit-filter: blur(8px);
  filter: blur(8px);
  z-index: 1100;
}

<div
  className="c0"
>
  <img
    alt=""
    className="c1"
    src="/images/responsive-sticker/small.png"
  />
</div>
`;

exports[`Responsive Image with progressive loading snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  display: block;
  height: 0;
  overflow: hidden;
  padding-top: 100%;
  position: relative;
  width: 100%;
}

.c1 {
  height: auto;
  left: 50%;
  opacity: 1;
  position: absolute;
  -webkit-transition: opacity 300ms cubic-bezier(0.4,0,0.2,1);
  transition: opacity 300ms cubic-bezier(0.4,0,0.2,1);
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  -webkit-filter: blur(8px);
  filter: blur(8px);
  z-index: 1100;
}

<div
  className="c0"
>
  <img
    alt=""
    className="c1"
    src="/images/responsive-sticker/small.png"
  />
</div>
`;
